src/main/java/io/stargate/sgv2/jsonapi/service/resolver/sort/TableSortClauseResolver.java

```diff
@@ -0,0 +1,158 @@
+import io.stargate.sgv2.jsonapi.api.model.command.clause.sort.SortExpression;
+import io.stargate.sgv2.jsonapi.config.OperationsConfig;
+import io.stargate.sgv2.jsonapi.exception.SortException;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs.JSONCodecRegistry;
+import io.stargate.sgv2.jsonapi.service.operation.query.OrderByCqlClause;
+import io.stargate.sgv2.jsonapi.service.operation.tables.TableANNOrderByCQlClause;
+import io.stargate.sgv2.jsonapi.service.schema.tables.ApiColumnDef;
+import io.stargate.sgv2.jsonapi.service.schema.tables.ApiTypeName;
+import io.stargate.sgv2.jsonapi.util.CqlVectorUtil;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Resolves a sort clause to deterimine if we want to apply a CQL ORDER BY clause to the operation.
+ *
+ * <p>Not all sort clauses result in a CQL order by, some may result in memory sorting.
+ */
+public class TableSortClauseResolver<CmdT extends Command & Sortable>
+    extends SortClauseResolver<CmdT, TableSchemaObject> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TableSortClauseResolver.class);
+
+  public TableSortClauseResolver(
+      OperationsConfig operationsConfig, JSONCodecRegistry codecRegistry) {
+    super(operationsConfig);
+  }
+
+  @Override
+  public OrderByCqlClause resolve(CommandContext<TableSchemaObject> commandContext, CmdT command) {
+    Objects.requireNonNull(commandContext, "commandContext is required");
+    Objects.requireNonNull(command, "command is required");
+
+    // NOTE: Currently only supporting vector sort, next PR will deal with clustering key sorting
+    var apiTableDef = commandContext.schemaObject().apiTableDef();
+    var sortClause = command.sortClause();
+    if (sortClause == null || sortClause.isEmpty()) {
+      return OrderByCqlClause.NO_OP;
+    }
+
+    var vectorSorts = sortClause.tableVectorSorts();
+    if (vectorSorts.isEmpty()) {
+      // For now, only supporting vector sort
+      return OrderByCqlClause.NO_OP;
+    }
+
+    if (vectorSorts.size() > 1) {
+      throw SortException.Code.MORE_THAN_ONE_VECTOR_SORT.get(
+          errVars(
+              commandContext.schemaObject(),
+              map -> {
+                map.put(
+                    "vectorColumns",
+                    errFmtApiColumnDef(apiTableDef.allColumns().filterBy(ApiTypeName.VECTOR)));
+                map.put(
+                    "sortColumns",
+                    errFmtJoin(vectorSorts.stream().map(SortExpression::path).toList()));
+              }));
+    }
+
+    // we have one vector sort - cannot have any other sorting
+    var nonVectorSorts = sortClause.nonTableVectorSorts();
+    if (!nonVectorSorts.isEmpty()) {
+      throw SortException.Code.CANNOT_MIX_VECTOR_AND_NON_VECTOR_SORT.get(
```

Comment:
If you see sortClauseDeserializer, there are similar logic for collection path, although table and collection path are mixed in deseralizer....

      if (DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(path)) {
        // Vector search can't be used with other sort clause
        if (totalFields > 1) {
          throw ErrorCodeV1.VECTOR_SEARCH_USAGE_ERROR.toApiException();
        }
        if (!inner.getValue().isArray()) {
          throw ErrorCodeV1.SHRED_BAD_VECTOR_VALUE.toApiException();
        }

perhaps, we could move this validation into CollectionSortClauseResolver as well in the future, just to have cleaner Deserializer code.

Reply:
as discussed - trying to get this in without re-doing the SortClauseDeserialiser, it is messy because it has some validation in it that should be later on when we know if it is a table or collection